evap/evaluation/models.py

```diff
@@ -348,36 +361,22 @@ def can_user_vote(self, user):
     def can_user_see_course(self, user):
         if user.is_reviewer:
             return True
-        if self.is_user_contributor_or_delegate(user):
-            return True
-        if user in self.participants.all():
-            return True
-        if self.is_private:
-            return False
-        if user.is_external:
-            return False
+        if self.is_private or user.is_external:
+            if not self.is_user_contributor_or_delegate(user) and not self.participants.filter(pk=user.pk).exists():
+                return False
         return True
 
     def can_user_see_results_page(self, user):
+        if self.is_single_result:
+            return False
         if user.is_reviewer:
             return True
         if self.state != 'published':
             return False
-        if self.is_user_contributor_or_delegate(user):
-            return True
-        if not self.can_publish_rating_results:
-            return False
-        if self.semester.results_are_archived:
-            return False
-        return self.can_user_see_course(user)
-
-    @property
-    def is_single_result(self):
-        # early return to save some queries
-        if self.vote_start_datetime.date() != self.vote_end_date:
-            return False
-
-        return self.contributions.filter(responsible=True, questionnaires__name_en=Questionnaire.SINGLE_RESULT_QUESTIONNAIRE_NAME).exists()
+        if not self.can_publish_rating_results or self.semester.results_are_archived or not self.can_user_see_course(user):
+            if not self.is_user_contributor_or_delegate(user):
+                return False
```

Comment:
What do you think about `return self.is_user_contributor_or_delegate(user)`
Using the fallthrough for the other case seems more like a "we do that because it's easier to code" and not a "we do that because it's semantically more true.". So, it would be more precise (imo) and shorter